hazelcast/src/test/java/com/hazelcast/util/FilteringClassLoader.java

```diff
@@ -96,6 +95,28 @@ public InputStream getResourceAsStream(String name) {
         return delegatingClassLoader.loadClass(name);
     }
 
+    public boolean checkResourceExcluded(String resourceName) {
+        if (resourceName == null) {
+            return true;
+        }
+        // transform resource path as class name so we can check against excluded packages
+        String resourceAsClassName = resourceName.replace('/', '.').replace(".class", "");
+        for (String excludedPackage : excludePackages) {
+            if (resourceAsClassName.startsWith(excludedPackage)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void checkExcluded(String name) throws ClassNotFoundException {
+        for (String excludedPackage : excludePackages) {
```

Comment:
The complexity of the algorithm can be simplified. Currently it is O(n) with n being the number of excluded packages. It can be converted to O(1) by caching the calculation and use the name as key in the hashtable.